apps/commercetools/src/Editor/CategoryPreviews/CategoryPreviews.tsx

```diff
@@ -0,0 +1,79 @@
+import * as React from 'react';
+import arrayMove from 'array-move';
+import isEqual from 'lodash/isEqual';
+import difference from 'lodash/difference';
+import { FieldExtensionSDK } from 'contentful-ui-extensions-sdk';
+import { mapSort } from '../../utils';
+import { SortableList } from './SortableList';
+import { PreviewsFn, Product } from '../../interfaces';
+
+interface Props {
+  sdk: FieldExtensionSDK;
+  disabled: boolean;
+  onChange: (categories: string[]) => void;
+  categories: string[];
+  fetchCategoryPreviews: PreviewsFn;
+}
+
+interface State {
+  categoryPreviews: Product[];
+}
+
+export class CategoryPreviews extends React.Component<Props, State> {
+  state = {
+    categoryPreviews: []
+  };
+
+  componentDidMount() {
+    this.updateCategoryPreviews();
+  }
+
+  componentDidUpdate({ categories: prevCategories }: Props) {
+    if (!isEqual(this.props.categories, prevCategories)) {
+      const lengthHasChanged = this.props.categories.length !== prevCategories.length;
+      const categoriesDiffer = difference(this.props.categories, prevCategories).length > 0;
+      const shouldRefetchProducts = lengthHasChanged || categoriesDiffer;
```

Comment:
Would `difference` not pick up on a length change as well? Is the `lengthHasChanged` redundant?

Reply:
No, they cover difference scenarios in the end...

lodash difference: `Creates an array of array values not included in the other given arrays using SameValueZero for equality comparisons. The order and references of result values are determined by the first array.`

so if you delete an item from an array and compare it to another array that has the same items plus the one you deleted you get no difference, you need the length comparison too